diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BooleanExpressionsRule.kt

```diff
@@ -0,0 +1,208 @@
+        }
+        return correctedExpression
+    }
+
+    private fun fixBooleanExpression(
+        node: ASTNode,
+        simplifiedExpr: Expression<String>,
+        mapOfExpressionToChar: HashMap<String, Char>) {
+        var correctKotlinBooleanExpression = simplifiedExpr
+            .toString()
+            .replace("&", "&&")
+            .replace("|", "||")
+            .drop(1)  // dropping first (
+            .dropLast(1)  // dropping last )
+        mapOfExpressionToChar.forEach { (key, value) ->
+            correctKotlinBooleanExpression = correctKotlinBooleanExpression.replace(value.toString(), key)
+        }
+        node.replaceChild(node.firstChildNode, KotlinParser().createNode(correctKotlinBooleanExpression))
+    }
+
+    /**
+     * Checks if boolean expression can be simplified with distributive law.
+     *
+     * @return String? null if it cannot be simplified. Simplified string otherwise.
+     */
+    private fun checkDistributiveLaw(
+        expr: Expression<String>,
+        mapOfExpressionToChar: HashMap<String, Char>,
+        node: ASTNode): String? {
+        // checking that expression can be considered as distributive law
+        val commonDistributiveOperand = getCommonDistributiveOperand(node, expr.toString(), mapOfExpressionToChar)?.first() ?: return null
+        val correctSymbolsSequence = mapOfExpressionToChar.values.toMutableList()
+        correctSymbolsSequence.remove(commonDistributiveOperand)
+        correctSymbolsSequence.add(0, commonDistributiveOperand)
+        val expressionsLogicalOperator = expr.toString().first { it == '&' || it == '|' }
+        // we return expression depending on second operator
+        return returnNeededDistributiveExpression(expressionsLogicalOperator, correctSymbolsSequence)
+    }
+
+    /**
+     * Returns correct result string in distributive law
+     */
+    private fun returnNeededDistributiveExpression(firstLogicalOperator: Char, symbols: List<Char>): String {
+        val secondSymbol = if (firstLogicalOperator == '&') '|' else '&'  // this is used to alter symbols
+        val resultString = StringBuilder()
+        symbols.forEachIndexed { index, symbol ->
+            if (index == 0) {
+                resultString.append("$symbol $firstLogicalOperator (")
+            } else {
+                resultString.append("$symbol $secondSymbol ")
+            }
+        }
+        return resultString.delete(resultString.length - 2, resultString.length).append(")").toString()
+    }
+
+    /**
+     * Method that checks that the expression can be simplified by distributive law.
+     * Distributive law - A && B || A && C -> A && (B || C) or (A || B) && (A || C) -> A || (B && C)
+     *
+     * @return common operand for distributed law
+     */
+    private fun getCommonDistributiveOperand(
+        node: ASTNode,
+        expression: String,
+        mapOfExpressionToChar: HashMap<String, Char>): String? {
```

Comment:
Why does it return `String`, if all operands are actually `Char`s in this representation?